File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracket_dict = {
            ')': '(',
            '}': '{',
            ']': '['
        }
        stack = []
        open_brackets = bracket_dict.values()
        for char in s:
            if char in open_brackets:
                stack.append(char)
            elif len(stack)!=0 and bracket_dict[char]==stack[-1]:
                stack.pop()
            else:
                return False
        return len(stack) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        unordered_map<char, char> mappings {};
        mappings[')'] = '(';
        mappings['}'] = '{';
        mappings[']'] = '[';
        stack<char> stk {};
        for (char c : s) {
            if (c == '(' || c == '{' || c == '[') {
                stk.push(c);
            }
            else if (!stk.empty() && stk.top() == mappings[c]) {
                stk.pop();
            }
            else return false;
        }
        return stk.empty();
    }
};